### PR TITLE
Dynamic parameters

### PR DESCRIPTION
**Pull request by andrej-kolic-sm**
Add support for date expressions in parameter list

https://searchmetrics.atlassian.net/browse/IA-169

### CODE REVIEW

File: CollectApiData.js [hunk 1]
```diff
@@ -25,9 +25,11 @@ function datacollect() {
     };
 
     myConnector.getData = function (table, doneCallback) {
-		var data = JSON.parse(tableau.connectionData);
+		    var data = JSON.parse(tableau.connectionData);
 
-        var url_json = 'https://api.searchmetrics.com/v4/' + tableau.connectionName + '.json?' + data.params;
+		    const newParams = processExpressions(data.params);
```
Comment (andrej-kolic-sm) [inline CollectApiData.js:30]: here we replace expressions with real data. It's executed on every data fetch.

File: expressions.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+  var today = new Date();
+  var dateDiff = new Date(today.getFullYear(), today.getMonth() + monthDiff, today.getDate());
+  dateDiffStr = formatDate(dateDiff);
+
+  return dateDiffStr;
+}
+
+
+function formatDate(date) {
```
Comment (andrej-kolic-sm) [inline expressions.js:138]: format date object to 'YYYYMMDD' string

File: expressions.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
+
+//
+// expressions
+//
+
+const expressionsMap = {
```
Comment (andrej-kolic-sm) [inline expressions.js:6]: This is map of expressions and functions that are executed to replace them with real data

File: CollectApiData.js (hunk 1 above)
Comment (fernanda-alves-sm) [reply CollectApiData.js:30]: It worked well. Thank you Rey

**Review by fernanda-alves-sm — APPROVED** [review]

**Pull request merged**